### ISSUE TITLE
Performance is very bad on LLFF dataset with 2 input views

### ISSUE THREAD

**Issue by NagabhushanSN95** [mjmjeong/InfoNeRF#6]
Hi,

I tried your model on LLFF dataset with 2 input views. But the reconstruction of novel views is very bad. I've attached the images below (train viewpoints are reconstructed reasonably well). Is this expected with such fewer views or do you think something is going wrong?

I'm trying the fern scene. I used images 6 and 8 as train frames and images 5,7,9 as test frames.

Image 5  (test frame)
![0005](https://user-images.githubusercontent.com/16269001/191415648-5c01662a-c2df-40e8-8540-f5793d1416de.png)

Image 6  (train frame)
![0006](https://user-images.githubusercontent.com/16269001/191415655-0909bdec-e0e2-4b31-b4ce-dd940d9eb257.png)

Image 7  (test frame)
![0007](https://user-images.githubusercontent.com/16269001/191415680-81293bf1-2790-40f5-9bda-ec4a0e731ba0.png)

Image 8  (train frame)
![0008](https://user-images.githubusercontent.com/16269001/191415692-38f0aa2d-3a8a-48c0-bc22-fb581c36ca8d.png)

Image 9  (test frame)
![0009](https://user-images.githubusercontent.com/16269001/191415697-d08288b5-b933-4161-81d8-0fe9f26a5985.png)

Configs: `args.txt`
```
N_entropy = 1024
N_importance = 128
N_iters = 50000
N_rand = 1024
N_samples = 64
alpha_model_path = None
basedir = ../Runs/Training/Train0011
chunk = 32768
ckpt_render_iter = None
computing_entropy_all = False
config = None
datadir = ../Data/Databases/NeRF_LLFF/Data
dataset_type = NeRF_LLFF
debug = False
entropy = True
entropy_acc_threshold = 0.1
entropy_end_iter = None
entropy_ignore_smoothing = False
entropy_log_scaling = False
entropy_ray_lambda = 1
entropy_ray_zvals_lambda = 0.001
entropy_type = log2
eval_only = False
expname = fern
factor = 4
fewshot = 2
fewshot_seed = 0
ft_path = None
half_res = False
i_embed = 0
i_img = 500
i_print = 100
i_testset = 50000
i_video = 50000
i_wandb = 100
i_weights = 25000
lindisp = False
llffhold = 8
lrate = 0.0005
lrate_decay = 500
maskdir = None
multires = 10
multires_views = 4
near_c2w_rot = 5
near_c2w_trans = 0.1
near_c2w_type = rot_from_origin
netchunk = 65536
netdepth = 8
netdepth_fine = 8
netwidth = 256
netwidth_fine = 256
no_batching = True
no_coarse = False
no_ndc = False
no_reload = False
num_interpolation_poses = 3
perturb = 1.0
precrop_frac = 0.5
precrop_iters = 0
raw_noise_std = 0.0
render_factor = 0
render_mypath = False
render_only = False
render_pass = False
render_test = False
render_test_full = False
render_test_ray = False
render_train = False
shape = greek
smooth_pixel_range = None
smooth_sampling_method = near_pose
smoothing = False
smoothing_activation = norm
smoothing_end_iter = None
smoothing_lambda = 1
smoothing_rate = 1
smoothing_step_size = 5000
spherify = False
test_scene = None
testskip = 8
train_scene = None
train_set_num = 1
use_viewdirs = True
wandb = False
wandb_group = None
white_bkgd = False
```

**Comment by mjmjeong** [comment]
I used the entropy_ray_zvals_lambda = 0.01 for the LLFF dataset. (The LLFF experiment in the supplementary document did not use the information gain reduction loss.)
However, I recommend using information gain reduction loss for better results. 
Empirically, using `near_pixel` for `smooth_sampling_method` shows more stable results than `near_pose`.

```
# config file
smooth_sampling_method='near_pixel'
smooth_pixel_range=1 # should be optimized for your dataset (usually, the best value was 1)
smoothing_activation=softmax
smoothing_lambda=0.0001 # should be optimized for your dataset
smoothing_step = 2500

**Comment by NagabhushanSN95** [comment]
Thank you, I will try this. On a side note, the same configs (my first message above) did much better on the flower scene but worse on fortress scene. I'll try the configs you suggested.

**Comment by NagabhushanSN95** [comment]
I tried with the configs you suggested. Now after training, the rendered images are completely black. During training, loss decreased from 0.66 to 0.33, but PSNR never went above 5. 

Configs:
```
N_entropy = 1024
N_importance = 128
N_iters = 50000
N_rand = 1024
N_samples = 64
alpha_model_path = None
basedir = ../Runs/Training/Train0011
chunk = 32768
ckpt_render_iter = None
computing_entropy_all = False
config = None
datadir = ../Data/Databases/NeRF_LLFF/Data
dataset_type = NeRF_LLFF
debug = False
entropy = True
entropy_acc_threshold = 0.1
entropy_end_iter = None
entropy_ignore_smoothing = False
entropy_log_scaling = False
entropy_ray_lambda = 1
entropy_ray_zvals_lambda = 0.01
entropy_type = log2
eval_only = False
expname = fern
factor = 4
fewshot = 2
fewshot_seed = 0
ft_path = None
half_res = False
i_embed = 0
i_img = 500
i_print = 100
i_testset = 50000
i_video = 50000
i_wandb = 100
i_weights = 25000
lindisp = False
llffhold = 8
lrate = 0.0005
lrate_decay = 500
maskdir = None
multires = 10
multires_views = 4
near_c2w_rot = 5
near_c2w_trans = 0.1
near_c2w_type = rot_from_origin
netchunk = 65536
netdepth = 8
netdepth_fine = 8
netwidth = 256
netwidth_fine = 256
no_batching = True
no_coarse = False
no_ndc = False
no_reload = False
num_interpolation_poses = 3
perturb = 1.0
precrop_frac = 0.5
precrop_iters = 0
raw_noise_std = 0.0
render_factor = 0
render_mypath = False
render_only = False
render_pass = False
render_test = False
render_test_full = False
render_test_ray = False
render_train = False
shape = greek
smooth_pixel_range = 1
smooth_sampling_method = near_pixel
smoothing = False
smoothing_activation = softmax
smoothing_end_iter = None
smoothing_lambda = 0.0001
smoothing_rate = 1
smoothing_step_size = 2500
spherify = False
test_scene = None
testskip = 8
train_scene = None
train_set_num = 1
use_viewdirs = True
wandb = False
wandb_group = None
white_bkgd = False
```

Would enabling NDC cause any problem?

**Comment by city19992** [comment]
Could you please tell me how to implement infonerf on LLFF datasets? I did not find the file 'data_splits.pth' which is mentioned in run_nerf.py

`    if args.dataset_type == 'llff':
        data_info = torch.load('./data/nerf_llff_data/data_splits.pth')`

**Comment by mjmjeong** [comment]
Please check the updated checkpoints on LLFF and DTU [here](https://drive.google.com/drive/folders/1sw1uAmSVw1K3Mw76yRCk1jD7muQH9yVS).
